x-pack/test/api_integration/apis/infraops/metrics.js

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from 'expect.js';
+import { metricsQuery } from '../../../../plugins/infra/public/containers/metrics/metrics.gql_query';
```

Comment:
As a practice in graphql tests, maybe we want to locally define the query so the whole test is more readable and less entangled with client code?

We should be able to just inline something here like

```javascript
import gql from 'graphql-tag';

const metricsQuery = gql`
  ...
`;
```

Reply:
It does test the query which we're actually using from the UI, isn't this a good thing?

Reply:
In a way that's good, but I would almost consider that part of the e2e tests. To me it's very hard to answer the question "what aspect of the api does this test case cover?".

Maybe there's a way to have both: Have a test case like "should support the metric container query" per client-side query and other test cases that test specific things like "supports two metrics in one query" with a locally defined query.

Does that sound reasonable?

Reply:
Seems reasonable